app/src/main/java/com/kamron/pogoiv/logic/PokeInfoCalculator.java

```diff
@@ -312,4 +312,23 @@ private Pokemon getLowestEvolution(Pokemon poke) {
 
         return list;
     }
+
+    /**
+     * Get how much hp a pokemon will have at a certain level, including the stamina IV taken from the scan results.
+     * If the prediction is not exact because of big possible variation in stamina IV, the average will be returnred.
+     *
+     * @param ivScanResult    Scan results which includes stamina ivs
+     * @param selectedLevel   which level to get the hp for
+     * @param selectedPokemon Which pokemon to get Hp for
+     * @return An integer representing how much hp selectedpokemon with ivscanresult stamina ivs has at selectedlevel
+     */
+    public int getHPAtLevel(IVScanResult ivScanResult, double selectedLevel, Pokemon selectedPokemon) {
+        double lvlScalar = Data.CpM[(int) (selectedLevel * 2 - 2)];
```

Comment:
There's `Data.levelToLevelIdx(selectedLevel)` for this—but I should use this also for the existing code. Fixing this...